e2e/rook-test-framework/clients/k8s_rook_block.go

```diff
@@ -0,0 +1,138 @@
+package clients
+
+import (
+	"errors"
+	"github.com/rook/rook/e2e/rook-test-framework/contracts"
+)
+
+type k8sRookBlock struct {
+	transportClient contracts.ITransportClient
+}
+
+var (
+	listCmd              = []string{"rook", "block", "ls"}
+	wideDataToBlockPod   = []string{"sh", "-c", "WRITE_DATA_CMD"}
+	readDataFromBlockPod = []string{"cat", "READ_DATA_CMD"}
+)
+
+// Constructor to create k8sRookBlock - client to perform rook Block operations on k8s
+func CreateK8sRookBlock(client contracts.ITransportClient) *k8sRookBlock {
+	return &k8sRookBlock{transportClient: client}
+}
+
+// Function to create a Block using Rook
+// Input paramaters -
+//name - path to a yaml file that creates a pvc in k8s - yaml should describe name and size of pvc being created
+//size - not user for k8s implementation since its descried on the pvc yaml definition
+//Output - k8s create pvc operation output and/or error
+func (rb *k8sRookBlock) Block_Create(name string, size int) (string, error) {
+	cmdArgs := []string{name}
+	out, err, status := rb.transportClient.Create(cmdArgs, nil)
+	if status == 0 {
+		return out, nil
+	} else {
+		return err, errors.New("Unabale to create block")
+	}
+}
+
+// Function to delete a Block using Rook
+// Input paramaters -
+//name - path to a yaml file that where pvc is desirbed - delete is run on the the yaml definition
+//Output  - k8s delete pvc operation output and/or error
+func (rb *k8sRookBlock) Block_Delete(name string) (string, error) {
+	cmdArgs := []string{name}
+	out, err, status := rb.transportClient.Delete(cmdArgs, nil)
+	if status == 0 {
+		return out, nil
+	} else {
+		return err, errors.New("Unable to delete Block")
+	}
+}
+
+// Function to list all the blocks created/being managed by rook
+//Returns a ouput  for rook cli for block list
+func (rb *k8sRookBlock) Block_List() (string, error) {
+	out, err, status := rb.transportClient.Execute(listCmd, nil)
+	if status == 0 {
+		return out, nil
+	} else {
+		return err, errors.New("Unable to list all blocks")
+	}
+}
+
+// Function to map a Block using Rook
+// Input paramaters -
+//name - path to a yaml file that creates a pod  - pod should be defined to use a pvc that was created earlier
+//mountpath - not used in this impl since mountpath is defined in the pod definition
+//Output  - k8s create pod operation output and/or error
+func (rb *k8sRookBlock) Block_Map(name string, mountpath string) (string, error) {
```

Comment:
for k8s integration test no, mount path is not used. 
Our Goal is to come up with a common interface for running end to end tests in rook - it may be standalone,k8s or any other platform in future.  For the stand-alone scenario, we need the block name and mount path. K8s implementation of rook_block client ignores mount path but standalone(not yet implemented) will use it.